Seed database does not work correctly with in-memory database

The in-memory databse should always be seeded, regardless of `dll-auto` setting

Seeding is broken:

```
Caused by: org.hsqldb.HsqlException: data exception: string data, right truncation;  table: "assignment_task" column: "files"
	at org.hsqldb.error.Error.error(Unknown Source) ~[hsqldb-2.4.1.jar:2.4.1]
	at org.hsqldb.Table.enforceTypeLimits(Unknown Source) ~[hsqldb-2.4.1.jar:2.4.1]
	at org.hsqldb.Table.generateAndCheckData(Unknown Source) ~[hsqldb-2.4.1.jar:2.4.1]
	at org.hsqldb.Table.insertSingleRow(Unknown Source) ~[hsqldb-2.4.1.jar:2.4.1]
	at org.hsqldb.StatementDML.insertSingleRow(Unknown Source) ~[hsqldb-2.4.1.jar:2.4.1]
	at org.hsqldb.StatementInsert.getResult(Unknown Source) ~[hsqldb-2.4.1.jar:2.4.1]
	at org.hsqldb.StatementDMQL.execute(Unknown Source) ~[hsqldb-2.4.1.jar:2.4.1]
	at org.hsqldb.Session.executeCompiledStatement(Unknown Source) ~[hsqldb-2.4.1.jar:2.4.1]
	at org.hsqldb.Session.execute(Unknown Source) ~[hsqldb-2.4.1.jar:2.4.1]
	... 62 common frames omitted
Caused by: org.hsqldb.HsqlException: data exception: string data, right truncation
	at org.hsqldb.error.Error.error(Unknown Source) ~[hsqldb-2.4.1.jar:2.4.1]
	at org.hsqldb.error.Error.error(Unknown Source) ~[hsqldb-2.4.1.jar:2.4.1]
	at org.hsqldb.types.BinaryType.castOrConvertToType(Unknown Source) ~[hsqldb-2.4.1.jar:2.4.1]
	at org.hsqldb.types.BinaryType.convertToTypeLimits(Unknown Source) ~[hsqldb-2.4.1.jar:2.4.1]
	... 70 common frames omitted
```

--- a/src/main/kotlin/de/code_freak/codefreak/entity/AssignmentTask.kt
+++ b/src/main/kotlin/de/code_freak/codefreak/entity/AssignmentTask.kt
@@ -35,7 +35,7 @@ class AssignmentTask(
   /**
    * A tar archive of files that act as boilerplate code
    */
-  @Type( type = "binary" )
+  @Type( type = "image" )
   var files: ByteArray?,
 
   /**

--- a/src/main/kotlin/de/code_freak/codefreak/init/SeedDatabase.kt
+++ b/src/main/kotlin/de/code_freak/codefreak/init/SeedDatabase.kt
@@ -44,8 +44,11 @@ class SeedDatabase : ApplicationListener<ContextRefreshedEvent>, Ordered {
   @Value("\${spring.jpa.hibernate.ddl-auto:''}")
   private lateinit var schemaExport: String
 
+  @Value("\${spring.jpa.database:''}")
+  private lateinit var database: String
+
   override fun onApplicationEvent(event: ContextRefreshedEvent) {
-    if (!schemaExport.startsWith("create")) {
+    if (!schemaExport.startsWith("create") && database != "HSQL") {
       return
     }
 

--- a/src/test/kotlin/de/code_freak/codefreak/init/SeedDatabaseTest.kt
+++ b/src/test/kotlin/de/code_freak/codefreak/init/SeedDatabaseTest.kt
@@ -7,18 +7,20 @@ import org.junit.runner.RunWith
 import org.springframework.beans.factory.annotation.Autowired
 import org.springframework.boot.test.context.SpringBootTest
 import org.springframework.test.context.ActiveProfiles
+import org.springframework.test.context.TestPropertySource
 import org.springframework.test.context.junit4.SpringRunner
 
 @RunWith(SpringRunner::class)
 @SpringBootTest
 @ActiveProfiles("dev")
+@TestPropertySource(locations = ["/application-test.properties"])
 internal class CodeFreakApplicationTests {
 
   @Autowired
   private lateinit var assignmentRepository: AssignmentRepository
 
   @Test
-  fun `seed database is executed`() {
+  fun `seed database is executed in dev profile`() {
     assertTrue(assignmentRepository.findAll().count() > 0)
   }
 }

--- a/src/test/kotlin/de/code_freak/codefreak/init/SeedDatabaseTest.kt
+++ b/src/test/kotlin/de/code_freak/codefreak/init/SeedDatabaseTest.kt
@@ -1,0 +1,24 @@
+package de.code_freak.codefreak.init
+
+import de.code_freak.codefreak.repository.AssignmentRepository
+import org.junit.Assert.assertTrue
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.boot.test.context.SpringBootTest
+import org.springframework.test.context.ActiveProfiles
+import org.springframework.test.context.junit4.SpringRunner
+
+@RunWith(SpringRunner::class)
+@SpringBootTest
+@ActiveProfiles("dev")
+internal class CodeFreakApplicationTests {
+
+  @Autowired
+  private lateinit var assignmentRepository: AssignmentRepository
+
+  @Test
+  fun `seed database is executed`() {
+    assertTrue(assignmentRepository.findAll().count() > 0)
+  }
+}

--- a/src/test/kotlin/de/code_freak/codefreak/init/SeedDatabaseTest.kt
+++ b/src/test/kotlin/de/code_freak/codefreak/init/SeedDatabaseTest.kt
@@ -14,7 +14,7 @@ import org.springframework.test.context.junit4.SpringRunner
 @SpringBootTest
 @ActiveProfiles("dev")
 @TestPropertySource(locations = ["/application-test.properties"])
-internal class CodeFreakApplicationTests {
+internal class SeedDatabaseTest {
 
   @Autowired
   private lateinit var assignmentRepository: AssignmentRepository

--- a/src/test/resources/application-test.properties
+++ b/src/test/resources/application-test.properties
@@ -1,4 +1,4 @@
 spring.jpa.database=HSQL
 spring.jpa.show-sql=false
 
-spring.jpa.hibernate.ddl-auto=create
+spring.jpa.hibernate.ddl-auto=update

